src/modules/ekf2/EKF/control.cpp

```diff
@@ -187,6 +187,10 @@ void Ekf::controlFusionModes(const imuSample &imu_delayed)
 		_tas_data_ready = _airspeed_buffer->pop_first_older_than(imu_delayed.time_us, &_airspeed_sample_delayed);
 	}
 
+	if (_beacon_range_buffer) {
+		_beacon_range_data_ready = _beacon_range_buffer->pop_first_older_than(imu_delayed.time_us, &_beacon_range_sample_delayed);
```

Comment:
You can directly pop the data from inside the `controlBeaconRangeFusion` function since you don't need it anywhere else (see https://github.com/PX4/PX4-Autopilot/blob/db539d15bd964484aa1e6f29bcdf8f9e0723bb70/src/modules/ekf2/EKF/baro_height_control.cpp#L52 for example). Then you don't need to save a potentially outdated copy in `_beacon_range_sample_delayed`